### PR TITLE
bpo-37444: Update differing exception between builtins and importlib

### PR DESCRIPTION
**Pull request by nsiregar**
Imports now raise `TypeError` instead of `ValueError` for relative import failures. This makes things consistent between `builtins.__import__` and `importlib.__import__` as well as using a more natural import for the failure.

https://bugs.python.org/issue37444

Automerge-Triggered-By: @brettcannon

### CODE REVIEW

File: Lib/importlib/util.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def resolve_name(name, package):
     if not name.startswith('.'):
         return name
     elif not package:
-        raise ValueError(f'no package specified for {repr(name)} '
-                         '(required for relative module names)')
+        raise ImportError(f'no package specified for {repr(name)} '
```
Comment (tirkarthi) [inline Lib/importlib/util.py:32]: Does this need a documentation update in https://github.com/python/cpython/blob/master/Doc/library/importlib.rst for `resolve_name` regarding change in exception type?

Comment (nsiregar) [reply Lib/importlib/util.py:32]: Thank you, I've updated mentioned documentation.

File: Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Update differing exception between buildints and importlib
```
Comment (aeros) [inline Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst]: Corrects spelling of builtins, specify dunder methods, and add inline markup
```suggestion
Update differing exception between :meth:`builtins.__import__` and
:meth:`importlib.__import__`.
```

File: Lib/importlib/util.py (hunk 1 above)
Comment (brettcannon) [reply Lib/importlib/util.py:32]: And what does the language reference say (if anything)?

File: Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst (hunk 2 above)
Comment (brettcannon) [reply Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst]: You  actually don't need the parentheses when you specify `:meth:`; Sphinx will add them for you.

Comment (aeros) [reply Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst]: Oh good to know, thanks. I updated the suggestion accordingly.

File: Lib/importlib/util.py (hunk 1 above)
Comment (nsiregar) [reply Lib/importlib/util.py:32]: I can't find reference that mention about raising `ValueError` in here https://github.com/python/cpython/blob/master/Doc/reference/import.rst or perhaps I am looking in wrong place?

File: Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst (hunk 2 above)
Comment (nsiregar) [reply Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst]: Thanks, I've updated it

File: Doc/library/importlib.rst [hunk 3]
```diff
@@ -1433,9 +1433,9 @@ an :term:`importer`.
...
    check to see if the **package** argument is needed.
 
-   :exc:`ValueError` is raised if **name** is a relative module name but
+   :exc:`ImportError` is raised if **name** is a relative module name but
    package is a false value (e.g. ``None`` or the empty string).
-   :exc:`ValueError` is also raised a relative name would escape its containing
+   :exc:`ImportError` is also raised a relative name would escape its containing
    package (e.g. requesting ``..bacon`` from within the ``spam`` package).
 
    .. versionadded:: 3.3
```
Comment (ncoghlan) [inline Doc/library/importlib.rst:1441]: We'll want a `versionchanged` note here, similar to the existing note on `find_spec` below.

File: Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Update differing exception between :meth:`builtins.__import__` and :meth:`importli.__import__`.
```
Comment (ncoghlan) [inline Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst]: Typo - second reference should be to `importlib`

**Review by ncoghlan — COMMENTED** [review]
Just a couple of minor documentation tweaks needed before this is ready to merge.

File: Doc/library/importlib.rst (hunk 3 above)
Comment (nsiregar) [reply Doc/library/importlib.rst:1441]: Thanks, I've added `versionchanged` note.

File: Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst (hunk 4 above)
Comment (nsiregar) [reply Misc/NEWS.d/next/Core and Builtins/2019-07-20-22-34-42.bpo-37444.UOd3Xs.rst]: Thanks, I've updated it

**Review by ncoghlan — APPROVED** [review]
Thanks @nsiregar! I went ahead and added some What's New entries directly, as it's possible there may be code out there that's calling this API directly and expecting ValueError.

File: Doc/library/importlib.rst [hunk 5]
```diff
@@ -1433,13 +1433,18 @@ an :term:`importer`.
    ``importlib.util.resolve_name('sys', __package__)`` without doing a
    check to see if the **package** argument is needed.
 
-   :exc:`ValueError` is raised if **name** is a relative module name but
+   :exc:`ImportError` is raised if **name** is a relative module name but
    package is a false value (e.g. ``None`` or the empty string).
```
Comment (brettcannon) [inline Doc/library/importlib.rst]:
```suggestion
   **package** is a false value (e.g. ``None`` or the empty string).
```

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
I added a separate entry in What's New for the changes to `builtins.__import__` and fixed a doc bug that I noticed on an unchanged line, otherwise LGTM! Thanks, @nsiregar !

I've set the PR to automerge once CI passes. If it doesn't then just let us know, @nsiregar .

**Comment by nsiregar** [comment]
thanks @brettcannon , looks like CI is failing during documentation build and `make patchcheck`.

**Pull request merged**